### PR TITLE
Use Poetry in build scripts and CI

### LINKED ISSUE #6: CI for publishing pypgstac
I noticed that the latest version pypgstac on pypi is 0.2.4. The main branch has a version of 0.2.3, and there's not tags for all released versions.

We should set up GitHub Actions to do the publishing, much like we are doing in [pystac](https://github.com/stac-utils/pystac/blob/main/.github/workflows/release.yml). That way releases happen on each GitHub release, and the python package is published by CI against tagged versions. This will require the `stacutils` user to be made a maintainer on the pypgstac pypi repository.

### LINKED ISSUE #19: Push to dockerhub during CI publish workflow
As follow up work on #18, we need to add the ability to push docker images to the cipublish script. Pypi publishing is already handled by cipublish, and the addition of publishing docker images will take care of most of the functionality in the `Makefile`, which should be removed once this and any other functionality is covered by `scripts/`

### LINKED ISSUE #21: Integrate poetry into development environment scripts
As follow on work to #18, we need to integrate poetry into the development environment setup, which was left as follow-up work.

### PR DESCRIPTION
**Pull request by bitner**
Remove remnants (setup.py, requirements-dev.txt) and use Poetry for all builds and for pushing to PyPi.
Fixes #21 

Add release trigger for git tags that creates a release from Tags and also posts docker image to ghcr.io 
Fixes #25 
Fixes #19
Fixes #6

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -20,11 +20,11 @@ jobs:
       - name: Install release dependencies
         run: |
           python -m pip install --upgrade pip
-          pip install setuptools wheel twine
+          pip install poetry
```
Comment (kylebarron) [inline .github/workflows/release.yml:23]: Might make sense to lock the version of poetry used here?

**Review by kylebarron — APPROVED** [review]

**Comment by bitner** [comment]
I've merged in the work from #25 into this PR.

A push of the v0.2.8 tag from this branch successfully pushed pypgstac 0.2.8 up to pypi as well as a v0.2.8 docker image up to ghcr.io

**Comment by bitner** [comment]
@lossyrob The release process triggered by pushing this tag pushed your work from #29 up to PyPi and ghcr.io docker.

**Comment by lossyrob** [comment]
> The release process triggered by pushing this tag pushed your work from #29 up to PyPi and ghcr.io docker.

siiiick. Love when things work as intended :-)

**Comment by bitner** [comment]
Hey @lossyrob can I get a review posted on this so I can merge it?

**Review by lossyrob — APPROVED** [review]

**Pull request merged**